### DISCUSSION TITLE
Einspeisung wird als Bezug gezählt.

### DISCUSSION THREAD

**Discussion by Farquarl** [hoylabs/OpenDTU-OnBattery#1141] [Q&A] [answered]
Hallo Liebe Community,

mir ist etwas Seltsames aufgefallen. Es gibt Situationen wo die Einspeisung meines Kraftwerkes als Bezug gewertet wird.
Es passiert immer dann wenn es einen hohen Verbrauch gibt. Ich nutze einen Tasmota Lesekopf, nutze also die vom Stromzähler bereitgestellten Werte.
Wenn jetzt der Bezug zB bei der Nutzung von einer Herdplatte ansteigt, liefert der Hoymiles 800w.
Wird jetzt die Herdplatte ausgeschaltet liefert der Wechselrichter mehr Energie als benötigt. Der Stromzähler wertet dies anscheinend als Verbrauch. Der nächste Wert in der DTU ist dann aber Einspeisung minus Grundlast und die DTU versucht das auszugleichen. Das führt dann dazu das immer 800w eingespeißt werden und ich den Überschuss praktisch bezahle.
Hat jemand ne Idee?
![grafik](https://github.com/user-attachments/assets/d28ec718-d0ca-4617-9e38-c084a656ac0f)

**Comment by spcqike** [comment]
Sicher, dass du einen „Always positive“ Zähler hast? Dieser hätte dieses Symbol 

<img width="98" alt="image" src="https://github.com/user-attachments/assets/97fb9b06-93c7-4d03-a02f-c9713cf192d5">

und er würde alle Einspeisungen als „Verbrauch“ zählen. Was passiert denn, wenn du den DPL und den wechselrichter einmal deaktivierst und du manuell mal etwas mehr als deinen Verbrauch einstellst. Sagen wir 250W Einspeisung bei 200W Verbrauch ohne wechselrichter. 

Ein always positiver Zähler würde dann 50W Verbrauch anzeigen. (Oder saldiert der nicht „richtig“ und könnte im worstcase sogar bereits 450W Verbrauch melden?)

**Comment by Farquarl** [comment]
Das ist ja das komische. Es ist kein "always positive" aber er verhält sich genau so.
Beispiel:

1. Verbrauch 500W (lt, Tasmota)
2. Hoymiles 800w Einspeisung.
3. Manueller Neustart des Hoymiles.
4. Verbrauch sinkt auf 300W und DTU regelt ganz normal auf +20w (habe ich so eingestellt.)
5. Homyiles liefert jetzt 280w.

**Comment by Farquarl** [comment]
Ich schätze ich kann nur auf ein shelly wechseln der auch die Richtung sieht. Das müsste funktionieren oder?

**Reply by spcqike** [reply]
Ja. Sollte. 

Allerdings frage ich mich dann, was und wie dein abrechnungszähler zählt. 

0W Saldo am Shelly, hin oder her. Wenn der Zähler vom Netzbetreiber dann trotzdem einen Verbrauch anzeigt und vielleicht sogar zählt, wäre das blöd. 

Check das lieber vorher. Keine Ahnung, Zählerstand notieren und mit „500W“ inkl. Einspeisung laufen lassen. Stunde später gucken ob du 0.5kWh mehr hast oder eben nicht. 

Sollte der das auch so falsch zählen, würd ich den Zähler tauschen lassen.

**Reply by Farquarl** [reply]
Mach ich mal, aber ich bin ziemlich sicher das er den Strom als Verbrauch zählt. Der Zähler kann ja nur ein Richtung und was tasmota anzeigt steht so auch 1 zu 1 auf dem Zähler.
Super ärgerlich die Geschichte. Zählerplatz plus unterverteilung muss sowieso neu wegen Wärmepumpe und großer Pv auf dem Dach die nächsten Jahre.

**Comment by Artikel14** [comment]
Seltsam. Manchmal hatte ich auch schon den Eindruck, 
dass unser Digitalzähler Einspeisung als Verbrauch zählt.
Zählertyp ISKRA175 
Solar-WR: Hoymiles
Aber ich kann es noch nicht beweisen.

**Comment by Farquarl** [comment, marked as answer]
Ich habe das Problem gefunden. Es ist anscheinend so das mein Stromzähler ein special command für das tasmota scrip braucht. Ansonsten sind die Werte invertiert:

M,=so1 | special SML option for meters that use a bit in the status register to sign import or export like ED300L, AS2020 or DTZ541e.g. 1,=so1,00010800,65,11,65,11,00100700 for DTZ5411. obis code that holds the direction bit, 2. Flag identifier, 3. direction bit, 4. second Flag identifier (some meters use 2 different flags), 5. second bit, 6 obis code of value to be inverted on direction bit.
